plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/GeneralUtils.java

```diff
@@ -829,4 +847,56 @@ public static UUID getMixedEndianUUIDFromBytes(byte[] bytes) {
         target.rewind();
         return new UUID(target.getLong(), target.getLong());
     }
+
+    /**
+     * Find index of item in set
+     */
+    public static <T> int getIndexInSet(Set<T> set, @Nullable T value) {
+        if (value == null) {
+            return -1;
+        }
+        int result = 0;
+        for (T t : set) {
+            if (t.equals(value)) {
+                return result;
+            }
+            result++;
+        }
+        return -1;
+    }
+
+    /**
+     * Store system encoding specification
+     */
+    public static void storeSystemEncoding(String encoding) {
+        systemFileEncoding = encoding;
+    }
+
+    /**
+     * Return available charsets with first item of system encoding
+     */
+    public static List<String> availableCharsets() {
+        List<String> charsetList = new ArrayList<>();
+        charsetList.add(String.format("%s (%s)", "Default encoding", systemFileEncoding));
+        charsetList.addAll(Charset.availableCharsets().keySet());
+        return charsetList;
+    }
+
+    /**
+     * Return index of charset specification
+     */
+    public static int getIndexOfAvailableCharset(String encoding) {
+        return availableCharsets().indexOf(encoding);
+    }
+
+    /**
+     * Return charset specification by index
+     */
+    public static String getCharsetByIndex(int index) {
```

Comment:
This too. Just use encoding name from combobox text

Reply:
Done